ui/src/middlewares/searchDispatcher.js

```diff
@@ -19,32 +19,38 @@ export default function({ getState, dispatch }) {
     if (
       action.type === LOCATION_CHANGE &&
       action.payload &&
-      action.payload.search &&
-      !action.payload.pathname.startsWith(SUBMISSIONS)
+      action.payload.location.search &&
+      !action.payload.location.pathname.startsWith(SUBMISSIONS)
     ) {
       const currentLocation = getLocationFromRootState(getState());
       const result = next(action);
-      const nextLocation = action.payload; // use this instead of `getLocationFromRootState(getState())` for testability.
+      const nextLocation = action.payload.location; // use this instead of `getLocationFromRootState(getState())` for testability.
+      const { isFirstRendering } = action.payload;
+      const hasPathnameChanged =
+        nextLocation.pathname !== currentLocation.pathname;
       if (
-        nextLocation.search !== currentLocation.search ||
-        nextLocation.pathname !== currentLocation.pathname
+        isFirstRendering ||
+        hasPathnameChanged ||
+        nextLocation.search !== currentLocation.search
```

Comment:
I know you don't use this anywhere else but it would be nice to put it in hasSearchChanged for readability